server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -131,15 +133,52 @@ public boolean isEnabled() {
     /**
      * The least amount of time data should be kept by elasticsearch.
      * @return the time period or null, null represents that data should never be deleted.
+     * @deprecated use {@link #getEffectiveDataRetention(DataStreamGlobalRetention)}
      */
+    @Deprecated
     @Nullable
     public TimeValue getEffectiveDataRetention() {
-        return getDataStreamRetention();
+        return getEffectiveDataRetention(null);
+    }
+
+    /**
+     * The least amount of time data should be kept by elasticsearch.
+     * @return the time period or null, null represents that data should never be deleted.
+     */
+    @Nullable
+    public TimeValue getEffectiveDataRetention(@Nullable DataStreamGlobalRetention globalRetention) {
+        return getEffectiveDataRetentionWithSource(globalRetention).v1();
+    }
+
+    /**
+     * The least amount of time data should be kept by elasticsearch.
+     * @return the time period or null, null represents that data should never be deleted.
+     */
+    @Nullable
```

Comment:
It doesn't look like this can actually ever return null, can it? Also, maybe best to make this package private? The only reason it needs to be exposed at all is for a unit test (and do we even need the source there since it doesn't seem to be used in any production code, or could it just use the method above (I haven't read the test yet so maybe it will become clear then)?)

Reply:
After looking at the test, I think we could test the same things without using the RetentionSource at all. Then we could get rid of this method and the RetentionSource enum, and just test `getEffectiveDataRetention` directly, right? Or is this something that's going to be used in a future PR?

Reply:
It's something coming in a future PR. When we return the lifecycle in a few GET APIs, the lifecycle section will the following structure:
```
{
  "enabled": true,
  "data_retention": "365d",
  "effective_retention": "90d",
  "retention_determined_by": "max_global_retention"
}
``` 
I thought it makes sense to calculate them together always to avoid duplicating this logic.